include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -2048,6 +2048,9 @@ inline ShapeInferContext::Ints ShapeInferContext::GetAttrInts(const char* attr_n
     Ort::ThrowOnError(ort_api_->ReadOpAttr(attr, ORT_OP_ATTR_INTS, ints.data(), out, &out));
     return ints;
   } else {
+    if (out == 0u) {
+      return {};
+    }
     return {i};
```

Comment:
Is it ever necessary to return this (`{0}`)? I'm wondering if we should just return `{}` if the `status` is false?

Reply:
bump

Reply:
@zz002 @mingyueliuh  
can this comment be addressed?

Reply:
There are two cases when status is nullptr; out = 1 means a number was read, in which case we return {i}; out = 0 means no number was read, in which case we return {}

Reply:
I see. I think I understand what is happening and I agree that your fix is correct.

A few notes:
- A `status == nullptr` means that `ReadOpAttr` was **successful**. A `status != nullptr` means failure.
- The `ReadOpAttr` function should normally be called twice: once to get the needed buffer size (returns a status != nullptr), and a second time to actually read the ints (returns status == null on success).
- This code tries a subtle optimization in the first call to `ReadOpAttr`. It passes in a buffer (`&i`) of size 1 just in case there is only 1 int. In this case, `status == nullptr` and we need to return `{i}`.

I think the code would be more understandable if we didn't try to do that last subtle optimization.

I'm comfortable approving the PR now, but, if possible, could you please leave the above list of observations as a comment above the first call to `ReadOpAttr`. It could save other developers a lot of time.

Reply:
Done ！